### PR TITLE
feat: custom dictionaries

### PR DESCRIPTION
**Pull request by DevinCodes**
| Q                 | A
| ----------------- | ----------
| Bug fix?          | no
| New feature?      | yes
| BC breaks?        | no     
| Need Doc update   | yes

## Describe your change
This adds the custom dictionaries feature along with tests.

### CODE REVIEW

File: src/SearchClient.php [hunk 1]
```diff
@@ -450,4 +451,153 @@ public function hasPendingMappings($requestOptions = array())
...
+     * Save entries to the given dictionary.
+     *
+     * @param string                      $dictionary
+     * @param array<array<string, mixed>> $entries
+     * @param bool                        $clearExistingDictionaryEntries
+     * @param array|RequestOptions        $requestOptions
+     *
+     * @return DictionaryResponse
+     */
+    public function saveDictionaryEntries($dictionary, $entries, $clearExistingDictionaryEntries = false, $requestOptions = array())
```
Comment (chloelbn) [inline src/SearchClient.php:465]: `$clearExistingDictionaryEntries` shouldn't be passed as an argument

File: src/SearchClient.php [hunk 2]
```diff
@@ -450,4 +451,153 @@ public function hasPendingMappings($requestOptions = array())
...
+    {
+        $response = $this->api->write(
+            'POST',
+            api_path('/1/dictionaries/%s/batch', $dictionary),
+            array(
+                'clearExistingDictionaryEntries' => $clearExistingDictionaryEntries,
+                'requests' => array_map(function ($entry) {
+                    Helpers::ensureObjectID($entry, 'ObjectID is required to add a dictionary entry.');
```
Comment (chloelbn) [inline src/SearchClient.php:473]: Could you make that logic into a new method?

File: src/SearchClient.php [hunk 3]
```diff
@@ -450,4 +451,153 @@ public function hasPendingMappings($requestOptions = array())
...
+     *
+     * @param string                      $dictionary
+     * @param array<array<string, mixed>> $entries
+     * @param array                       $requestOptions
+     *
+     * @return DictionaryResponse
+     */
+    public function replaceDictionaryEntries($dictionary, $entries, $requestOptions = array())
+    {
+        return $this->saveDictionaryEntries(
```
Comment (chloelbn) [inline src/SearchClient.php:498]: here you can call the endpoint again with `$clearExistingDictionaryEntries` hard-coded to `true`

**Review by chloelbn — CHANGES_REQUESTED** [review]

File: src/Support/Helpers.php [hunk 4]
```diff
@@ -126,4 +126,16 @@ public static function mapObjectIDs($objectIDKey, $objects)
             return $object;
         }, $objects);
     }
+
+    public static function buildDictionaryRequests(array $entries, $action)
```
Comment (chloelbn) [inline src/Support/Helpers.php:130]: you're not using it in the methods but `buildBatch`

**Review by chloelbn — CHANGES_REQUESTED** [review]

File: src/Support/Helpers.php (hunk 4 above)
Comment (DevinCodes) [reply src/Support/Helpers.php]: Good catch, I completely forgot to remove this 😅

**Review by chloelbn — APPROVED** [review]

**Pull request merged**